### PR TITLE
Create `Edit` and `CodeSnippet` components

### PR DESCRIPTION
**Pull request by katydecorah**
This PR adds a new component that you can send code and metadata to and these buttons will bring it into your favorite editor:

image

This PR also adds CodeSnippet that include:
- option to add Edit in buttons
- option add CodeSnippetTitle
- automatically pulls in prism.css

To do:
* [x] more tests, especially to catch errors with prettier attempting to parse bad html
* [x] break code into smaller functions, add tests
* [x] rework props to simplify (html, css, js, resources, frontmatter) we are currently doing a lot of massaging and pieces together of data to get it how we want, instead we should provide helper functions to help extract data so that this component can stay simple

## How to test /mapbox-gl-js-docs:
1. from this branch, `npm run build` (build it locally)
2. in mapbox-gl-js-docs, pull down the `dr-ui-edit-buttons` branch
3. in mapbox-gl-js-docs, install dr-ui locally: `"@mapbox/dr-ui": "file:/Users/katydecorah/Documents/GitHub/dr-ui/pkg",` (update this with your path to your local dr-ui)
4. in mapbox-gl-js-docs, `yarn install` then `yarn start`
5. navigate to an examples page and test any of the edit buttons

## How to test in /help:

1. from this branch, `npm run build` (build it locally)
2. in help, pull down the `dr-ui-code-snippet` branch
3. in help, install dr-ui locally: `"@mapbox/dr-ui": "file:/Users/katydecorah/Documents/GitHub/dr-ui/pkg",` (update this with your path to your local dr-ui)
4. in help, `npm install` then `npm start`
5. navigate to http://localhost:8080/help/tutorials/use-mapbox-gl-js-with-react/  and see the CodeSnippet in action

### CODE REVIEW

**Comment by katydecorah** [comment]
This is ready for review, but can wait until the new year.

**Comment by katydecorah** [comment]
Per chat, we're going to work in mr-ui code-snippet to this PR

**Comment by katydecorah** [comment]
Need to test locally in:

* [ ] mapbox-gl-js-docs
* [ ] help

File: src/components/code-snippet/code-snippet.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
+/* eslint-disable xss/no-mixed-html */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import MrCodeSnippet from '@mapbox/mr-ui/code-snippet';
```
Comment (colleenmcginnis) [inline src/components/code-snippet/code-snippet.js]: `MrCodeSnippet` 😂

**Comment by katydecorah** [comment]
@colleenmcginnis this is ready for review again! I update the original comment with some testing suggestions

File: src/components/code-snippet/__tests__/code-snippet-test-cases.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  component: CodeSnippet,
+  description: 'A CodeSnippet with every feature (edit buttons, title)',
+  props: {
+    code: `<h1>hi</h1>`,
+    highlighter: () => highlightHtml,
+    filename: 'index.html',
+    edit: {
+      html: '<h1>gurd murn!</h1>',
+      css: 'h1 {color: red}',
+      js: "console.log('hurn')",
```
Comment (colleenmcginnis) [inline src/components/code-snippet/__tests__/code-snippet-test-cases.js]: Trying to wrap my head around a real world example here. So if we were to use this with a tutorial code block, we could import the `code` and then use the `extractor` helper function to get the `edit` contents? Like:

```
<CodeSnippet
  code={rawCode}
  highlighter={() => highlightHtml}
  edit={extractor(code)}
  ...
/>
```

Comment (katydecorah) [reply src/components/code-snippet/__tests__/code-snippet-test-cases.js]: I'll add a test case!

Comment (katydecorah) [reply src/components/code-snippet/__tests__/code-snippet-test-cases.js]: @colleenmcginnis added in https://github.com/mapbox/dr-ui/pull/197/commits/571610ddb2a221ec0abbbe51768c756e1842d426

**Pull request merged**